Modules/_ssl.c

```diff
@@ -2200,6 +2200,70 @@ _ssl__SSLSocket_group_impl(PySSLSocket *self)
 #endif
 }
 
+/*[clinic input]
+@critical_section
+_ssl._SSLSocket.client_sigalg
+[clinic start generated code]*/
+
+static PyObject *
+_ssl__SSLSocket_client_sigalg_impl(PySSLSocket *self)
+/*[clinic end generated code: output=499dd7fbf021a47b input=a0d9696b5414c627]*/
+{
+#if OPENSSL_VERSION_NUMBER >= 0x30500000L
+    int ret;
+    const char *sigalg;
+
+    if (self->ssl == NULL) {
+        Py_RETURN_NONE;
+    }
+    if (self->socket_type == PY_SSL_CLIENT) {
+        ret = SSL_get0_signature_name(self->ssl, &sigalg);
+    } else {
+        ret = SSL_get0_peer_signature_name(self->ssl, &sigalg);
+    }
+    if (ret == 0) {
+        Py_RETURN_NONE;
+    }
```

Comment:
```suggestion
    ret = (self->socket_type == PY_SSL_CLIENT)
        ? SSL_get0_signature_name(self->ssl, &sigalg)
        : SSL_get0_peer_signature_name(self->ssl, &sigalg);
    if (ret == 0) {
        Py_RETURN_NONE;
    }
```

If you want a shorten version. If not, put `} else {` on separate lines (PEP-7).

Reply:
Thanks - I went with your suggested refactoring here.